src/tracks/playabletrack/wavetrack/ui/SampleHandle.cpp

```diff
@@ -385,8 +400,8 @@ UIHandle::Result SampleHandle::Drag
    const auto size = ( end - start + 1 ).as_size_t();
    if (size == 1) {
       // Don't require dithering later
-      mClickedTrack->Set(
-         (samplePtr)&newLevel, floatSample, start, size, narrowestSampleFormat);
+      constexpr auto iChannel = 0u;
+      mClickedTrack->SetFloatAt(t0, iChannel, newLevel, narrowestSampleFormat);
```

Comment:
I'm reviewing changes in Drag now, paying attention to what did not change as well as what did.

I mentioned before the two `sampleCount` typed members of SampleHandle.  They are 

```
   sampleCount mClickedStartSample{};
   sampleCount mLastDragSample{};
```

and they are used in drag.  But they depend on `LongSamplesToTime` and `TimeToLongSamples` which should be avoided now.

Change those two to store pixel x coordinates instead -- that is, `ViewInfo::TimeToPosition` values will quantize the time, suitable for a comparison of click position with drag position.

Reply:
Compute a time value at the top of Drag, not a sampleCount value s0.

Reply:
I have no more to say about the Sample handle.

It was good to review this -- and understand how pixel to time and inverse mappings will work with the new data structures, and to understand what's missing for envelope.